src/clients/kmd.ts

```diff
@@ -0,0 +1,235 @@
+import { providers } from "../providers";
+import type { Account, Wallet, WalletProvider } from "../types";
+import { TransactionsArray } from "../types";
+
+// Default config settings for dev sandbox kmd
+const DEFAULT_KMD_TOKEN = "a".repeat(64);
+const DEFAULT_KMD_HOST = "http://localhost";
+const DEFAULT_KMD_PORT = "4002";
+const DEFAULT_KMD_WALLET = "unencrypted-default-wallet";
+const DEFAULT_KMD_PASSWORD = "";
+
+type KMDConfig = {
+  host: string;
+  port: string;
+  token: string;
+};
+
+const DefaultKMDConfig = {
+  host: DEFAULT_KMD_HOST,
+  token: DEFAULT_KMD_TOKEN,
+  port: DEFAULT_KMD_PORT,
+} as KMDConfig;
+
+interface ListWalletResponse {
+  id: string;
+  name: string;
+  driver_name?: string;
+  driver_version?: number;
+  mnemonic_ux?: boolean;
+  supported_txs?: Array<any>;
+}
+
+interface InitWalletHandle {
+  wallet_handle_token: string;
+  message?: string;
+  error?: boolean;
+}
+
+type InitWallet = {
+  client: algosdk.Kmd;
+  id: PROVIDER_ID;
+  providers: typeof providers;
+};
+
+class KMDWallet extends BaseWallet {
+  #client: algosdk.Kmd;
+  walletId: string;
+  id: PROVIDER_ID;
+  provider: WalletProvider;
+
+  constructor(initAlgodClient: InitAlgodClient, initWallet: InitWallet) {
+    super(initAlgodClient);
+
+    this.#client = initWallet.client;
+    this.id = initWallet.id;
+    this.provider = initWallet.providers[this.id];
+    this.walletId = "";
+  }
+
+  static async init() {
+    const initAlgodClient: InitAlgodClient = {
+      algosdk,
+      token: NODE_TOKEN,
+      server: NODE_SERVER,
+      port: NODE_PORT,
+    };
+
+    // TODO: allow diff config options?
+    const kmdConfig: KMDConfig = DefaultKMDConfig;
+
+    const kmdClient = new algosdk.Kmd(
+      kmdConfig.token,
+      kmdConfig.host,
+      kmdConfig.port
+    );
+
+    const initWallet: InitWallet = {
+      id: PROVIDER_ID.KMD_WALLET,
+      client: kmdClient,
+      providers: providers,
+    };
+
+    return new KMDWallet(initAlgodClient, initWallet);
+  }
+
+  async connect(): Promise<Wallet> {
+    // TODO: prompt for wallet and password?
+    return {
+      ...this.provider,
+      accounts: await this.listAccounts(
+        DEFAULT_KMD_WALLET,
+        await this.requestPassword()
+      ),
+    };
+  }
+
+  async disconnect() {
+    return;
+  }
+
+  async reconnect(): Promise<Wallet | null> {
+    return {
+      ...this.provider,
+      accounts: await this.listAccounts(
+        DEFAULT_KMD_WALLET,
+        await this.requestPassword()
+      ),
+    };
+  }
+
+  async requestPassword(): Promise<string> {
+    // TODO: store it locally?
+    const pw = prompt("gib password");
+    return pw ? pw : "";
+  }
+
+  async getWalletToken(walletId: string, password: string): Promise<string> {
+    const handleResp: InitWalletHandle = await this.#client.initWalletHandle(
+      walletId,
+      password
+    );
+    return handleResp.wallet_handle_token;
```

Comment:
We may want to consider some persisted token that we can refresh on some interval rather than requesting a new one every time and requiring the user to type their password